server/src/main/java/org/apache/iotdb/db/metadata/mnode/InternalMNode.java

```diff
@@ -0,0 +1,315 @@
+  }
+
+  /** delete a child */
+  @Override
+  public void deleteChild(String name) {
+    if (children != null) {
+      children.remove(name);
+    }
+  }
+
+  /** delete the alias of a child */
+  @Override
+  public void deleteAliasChild(String alias) {
+    if (aliasChildren != null) {
+      aliasChildren.remove(alias);
+    }
+  }
+
+  @Override
+  public Template getDeviceTemplate() {
+    return deviceTemplate;
+  }
+
+  @Override
+  public void setDeviceTemplate(Template deviceTemplate) {
+    this.deviceTemplate = deviceTemplate;
+  }
+
+  /** get the child with the name */
+  @Override
+  public IMNode getChild(String name) {
+    IMNode child = null;
+    if (children != null) {
+      child = children.get(name);
+    }
+    if (child != null) {
+      return child;
+    }
+    return aliasChildren == null ? null : aliasChildren.get(name);
+  }
+
+  @Override
+  public IMNode getChildOfAlignedTimeseries(String name) throws MetadataException {
+    IMNode node = null;
+    // for aligned timeseries
+    List<String> measurementList = MetaUtils.getMeasurementsInPartialPath(new PartialPath(name));
+    for (String measurement : measurementList) {
+      IMNode nodeOfMeasurement = getChild(measurement);
+      if (node == null) {
+        node = nodeOfMeasurement;
+      } else {
+        if (node != nodeOfMeasurement) {
+          throw new AlignedTimeseriesException(
+              "Cannot get node of children in different aligned timeseries", name);
+        }
+      }
+    }
+    return node;
+  }
+
+  /** get the count of all MeasurementMNode whose ancestor is current node */
+  @Override
+  public int getMeasurementMNodeCount() {
+    if (children == null) {
+      return 0;
+    }
+    int measurementMNodeCount = 0;
+    for (IMNode child : children.values()) {
+      measurementMNodeCount += child.getMeasurementMNodeCount();
+    }
+    return measurementMNodeCount;
+  }
+
+  /** add an alias */
+  @Override
+  public boolean addAlias(String alias, IMNode child) {
+    if (aliasChildren == null) {
+      // double check, alias children volatile
+      synchronized (this) {
+        if (aliasChildren == null) {
+          aliasChildren = new ConcurrentHashMap<>();
+        }
+      }
+    }
+
+    return aliasChildren.computeIfAbsent(alias, aliasName -> child) == child;
+  }
+
+  @Override
+  public Map<String, IMNode> getChildren() {
+    if (children == null) {
+      return Collections.emptyMap();
+    }
+    return children;
+  }
+
+  @Override
+  public Map<String, IMNode> getAliasChildren() {
+    if (aliasChildren == null) {
+      return Collections.emptyMap();
+    }
+    return aliasChildren;
+  }
+
+  @Override
+  public void setChildren(Map<String, IMNode> children) {
+    this.children = children;
+  }
+
+  public void setAliasChildren(Map<String, IMNode> aliasChildren) {
+    this.aliasChildren = aliasChildren;
+  }
+
+  @Override
+  public void serializeTo(MLogWriter logWriter) throws IOException {
```

Comment:
serializeTo and deserializeFrom is relative，  if  you implement serializeTo, you will need add a deserializeFrom.

Reply:
Fixed

Reply:
It's just under the serializeChildren method.

Reply:
public static InternalMNode deserializeFrom(MNodePlan plan) {
    return new InternalMNode(null, plan.getName());
  }

Reply:
At line 259 in InternalMNode.java

Reply:
Only MLogReader uses MNode's deserializeFrom method, and the param is MNodePlan.